### PR TITLE
Implement neighborhood submissions & filtering

### PR DESCRIPTION
**Pull request by nachtm**
The neighborhoods filter now automatically populates from the endpoint created in #20 and #22. In addition, we can now filter on neighborhoods since some data points have neighborhoods! To get this working with redux, I had to modify `FilterService.getInitialFilter()` to take in a neighborhoods parameter, which defaults to `[]` in the initial state. Then, whenever we reset, we re-build the initial filter using the current filter's keys.

Also, the form now automatically gets neighborhoods from Mapbox's API. Take a look at some shots below:
All neighborhoods still shows all results (desktop):
image
But now you can filter by neighborhood (desktop):
image

Report view defaults to downtown:
image
Neighborhood changes as you drag the dot around:
image
And neighborhood defaults to "Unknown" when mapbox doesn't give us anything helpful:
image

Also, I've updated the filter list so it's sorted, but I took a screenshot before I had done so.

### CODE REVIEW

File: src/services/NeighborhoodService.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    getNeighborhoodFor = (lat, lng) => {
+        // Create request
+        return this.geocodingService.reverseGeocode({
+            query: [lng, lat],
+            types: ['neighborhood']
+        })
+            .send()
+            .then(response => {
+                const neighborhoods = response.body.features;
+                if (neighborhoods && neighborhoods.length > 0) {
```
Comment (sayas01) [inline src/services/NeighborhoodService.js]: Works perfectly! But how does the service determine which neighborhoods to look for? In other words, have you specified somewhere that we want Seattle neighborhood?

Comment (nachtm) [reply src/services/NeighborhoodService.js]: Nope -- we still have some validation to do here for sure. I wasn't sure what the extent of our range was. What do you think?

Comment (sayas01) [reply src/services/NeighborhoodService.js]: Oh I see! When I checked, I had not well zoomed in and was getting unknown for outside Seattle. lol! We could have a set of neighborhoods we are looking for and do a validation based on the set and the reverse geocoded value. I do not know if there is any helpful parameters in the map service that does this for us. I think we should check with Todd/Katie on the neighborhood limit.

Comment (nachtm) [reply src/services/NeighborhoodService.js]: I think based on the terms of use for this API, we wouldn't be allowed to get a list of all neighborhoods in the Seattle area according to Mapbox and use them as a whitelist. 

If we get some bounds from Todd/Katie/an analyst then we could ensure that the point is within the bounds as a first step before even sending the data to the API.

I agree that getting some boundaries would be a good next step!

**Pull request merged**